VeganVirus/VeganVirus/ExcelAction.cpp

```diff
@@ -0,0 +1,128 @@
+#include "ExcelAction.h"
+
+
+
+BOOL CALLBACK windowCountCallback(
+    HWND hwnd,
+    LPARAM lParam
+)
+{
+    WindowCountCallbackParams& params = *(WindowCountCallbackParams*)lParam;
+
+    char title[100] = { 0 };
+    GetWindowTextA(hwnd, title, 100);
+    char* ret;
+
+    // check if title contains params.searchTerm
+    ret = strstr(title, params.searchTerm);
+    if (ret != NULL) {
+        params.count++;
+    }
+    return TRUE;
+}
+ExcelAction::ExcelAction(double req) : Action(req, EXCEL_ICON_PATH)
+{
+}
+void ExcelAction::start()
+{
+    _preProcessCount = excelWindowCount();
+    this->activeFlag = true;
+}
+void ExcelAction::update(double dt)
+{
+    if (!this->activeFlag)
+    {
+        return;
+    }
+    int currPocessCount = excelWindowCount();;
+    if (currPocessCount > _preProcessCount)
+    {
+        createFile(num_file);
+        openFile(num_file);
+        _vectorExcelFiles.insert(num_file);
+        num_file += 1;
+        currPocessCount++;
+        Sleep(1000);
```

Comment:
Shouldn't have a sleep here. You can remove it, but it may cause problems. You can also implement a short timer to disable this action for 1 second.
(add a guard clause checking if the timer is on).